### PR TITLE
API/ENH: cull state PVs, include helpful motor PVs

### PR DESCRIPTION
**Pull request by ZLLentz**
## Description

Based on the discussions in https://docs.google.com/spreadsheets/d/1xfSB1mJwVu4tXvxBT85TEYVi29Z7VoQ4w5CXw13rV8A/edit#gid=0, make changes to pcdsdevices to:
- remove PVs that will be removed
- add PVs that were not removed that were kept because they are helpful (so, long overdue inclusions)

This includes the EPS classes added to the PLCs earlier this year.

## Motivation and Context

I want the python classes to match the overall IOCs as much as possible.

## How Has This Been Tested?

- Everything looks good on existing screens for the normal BeckhoffAxis case for all the motors I've checked.
- I couldn't find a live, running case to check for the old version of the class. Comment is below.
- I've adjusted the example PLC to include an EPS example for that variant.

## Where Has This Been Documented?

Just this PR so far. I'll make pre-release notes.

## Pre-merge checklist
- [x] Code works interactively
- [x] Code contains descriptive docstrings, including context and API
- [x] New/changed functions and methods are covered in the test suite where possible
- [x] Test suite passes locally
- [x] Test suite passes on GitHub Actions
- [x] Ran docs/pre-release-notes.sh and created a pre-release documentation page
- [x] Pre-release docs include context, functional descriptions, and contributors as appropriate

### CODE REVIEW

**Comment by ZLLentz** [comment]
I was unable to find a live PV that matches the BeckhoffAxis_Pre140 specification, but there are some pv lists that indicate that my choice here to blank out the posdiff and keep the other new PVs is correct:

<details>

```
$ grep_pv HXX:VON_HAMOS:MMS:01:PLC:
/reg/d/iocData/ioc-hxx-vonhamos/iocInfo
HXX:VON_HAMOS:MMS:01:PLC:bEnable, "bo"
HXX:VON_HAMOS:MMS:01:PLC:bReset, "bo"
HXX:VON_HAMOS:MMS:01:PLC:bExecute, "bo"
HXX:VON_HAMOS:MMS:01:PLC:nEnableMode_RBV, "mbbi"
HXX:VON_HAMOS:MMS:01:PLC:nBrakeMode_RBV, "mbbi"
HXX:VON_HAMOS:MMS:01:PLC:nHomingMode_RBV, "mbbi"
HXX:VON_HAMOS:MMS:01:PLC:fPosition, "ao"
HXX:VON_HAMOS:MMS:01:PLC:fVelocity, "ao"
HXX:VON_HAMOS:MMS:01:PLC:fAcceleration, "ao"
HXX:VON_HAMOS:MMS:01:PLC:fDeceleration, "ao"
HXX:VON_HAMOS:MMS:01:PLC:fHomePosition, "ao"
HXX:VON_HAMOS:MMS:01:PLC:sName_RBV, "waveform"
HXX:VON_HAMOS:MMS:01:PLC:sErrorMessage_RBV, "waveform"
HXX:VON_HAMOS:MMS:01:PLC:nEncoderCount_RBV, "longin"
HXX:VON_HAMOS:MMS:01:PLC:nCommand_RBV, "longin"
HXX:VON_HAMOS:MMS:01:PLC:nCmdData_RBV, "longin"
HXX:VON_HAMOS:MMS:01:PLC:nMotionAxisID_RBV, "longin"
HXX:VON_HAMOS:MMS:01:PLC:nErrorId_RBV, "longin"
HXX:VON_HAMOS:MMS:01:PLC:bLimitForwardEnable_RBV, "bi"
HXX:VON_HAMOS:MMS:01:PLC:bLimitBackwardEnable_RBV, "bi"
HXX:VON_HAMOS:MMS:01:PLC:bHome_RBV, "bi"
HXX:VON_HAMOS:MMS:01:PLC:bBrakeRelease_RBV, "bi"
HXX:VON_HAMOS:MMS:01:PLC:bHardwareEnable_RBV, "bi"
HXX:VON_HAMOS:MMS:01:PLC:bAllForwardEnable_RBV, "bi"
HXX:VON_HAMOS:MMS:01:PLC:bAllBackwardEnable_RBV, "bi"
HXX:VON_HAMOS:MMS:01:PLC:bAllEnable_RBV, "bi"
HXX:VON_HAMOS:MMS:01:PLC:bGantryForwardEnable_RBV, "bi"
HXX:VON_HAMOS:MMS:01:PLC:bGantryBackwardEnable_RBV, "bi"
HXX:VON_HAMOS:MMS:01:PLC:bPowerSelf_RBV, "bi"
HXX:VON_HAMOS:MMS:01:PLC:bGantryAxis_RBV, "bi"
HXX:VON_HAMOS:MMS:01:PLC:bEnable_RBV, "bi"
HXX:VON_HAMOS:MMS:01:PLC:bReset_RBV, "bi"
HXX:VON_HAMOS:MMS:01:PLC:bExecute_RBV, "bi"
HXX:VON_HAMOS:MMS:01:PLC:bEnableDone_RBV, "bi"
HXX:VON_HAMOS:MMS:01:PLC:bBusy_RBV, "bi"
HXX:VON_HAMOS:MMS:01:PLC:bDone_RBV, "bi"
HXX:VON_HAMOS:MMS:01:PLC:bSafetyReady_RBV, "bi"
HXX:VON_HAMOS:MMS:01:PLC:bError_RBV, "bi"
HXX:VON_HAMOS:MMS:01:PLC:nCommand, "longout"
HXX:VON_HAMOS:MMS:01:PLC:nCmdData, "longout"
HXX:VON_HAMOS:MMS:01:PLC:fPosition_RBV, "ai"
HXX:VON_HAMOS:MMS:01:PLC:fVelocity_RBV, "ai"
HXX:VON_HAMOS:MMS:01:PLC:fAcceleration_RBV, "ai"
HXX:VON_HAMOS:MMS:01:PLC:fDeceleration_RBV, "ai"
HXX:VON_HAMOS:MMS:01:PLC:fHomePosition_RBV, "ai"
```

</details>

**Comment by ZLLentz** [comment]
I'm taking a quick detour because it seems like one of the EPS PVs I've added here isn't building from lcls-twincat-general, possibly due to an error in the pytmc syntax- investigating.

I'm doing this detour because it's blocking me from fully testing this.

**Comment by ZLLentz** [comment]
In particular, the way the names interact with the rows in the detailed view obfuscates the source of an EPS problem. It looks a bit better in the tree view.
image

**Comment by ZLLentz** [comment]
This is now ready for review, but I don't want to merge it until https://github.com/pcdshub/lcls-twincat-motion/pull/192 is finalized. I will still mark as ready for review and ask for python reviewers.

File: pcdsdevices/state.py [hunk 1]
```diff
@@ -509,21 +509,10 @@ class TwinCATStateConfigOne(Device):
                      doc='The defined state name.')
     setpoint = Cpt(PytmcSignal, ':SETPOINT', io='io', kind='config',
                    doc='The corresponding motor set position.')
-    delta = Cpt(PytmcSignal, ':DELTA', io='io', kind='config',
-                doc='The deviation from setpoint that still counts '
-                    'as at the position.')
     velo = Cpt(PytmcSignal, ':VELO', io='io', kind='config',
                doc='Velocity to move to the state at.')
-    accl = Cpt(PytmcSignal, ':ACCL', io='io', kind='omitted',
```
Comment (tangkong) [inline pcdsdevices/state.py:517]: I'm surprised these weren't deemed important enough to keep, but I'm not the one using them.  

Is there a place where this information will live on, after its removal from here?  Would the PLC be the only place I can find out the `:ACCL` PV exists?

**Review by tangkong — APPROVED** [review]
Seems good to me, as long as everyone agrees on the removals

File: pcdsdevices/state.py (hunk 1 above)
Comment (ZLLentz) [reply pcdsdevices/state.py:517]: I've certainly had second thoughts about the removals- simply removing just the pmps info would be enough to "fix" the immediate problem and the rest might be enthusiastic overoptimization.

The PLC would be the only place that this info would be seen afterwards. So, you'd be unable to inspect or change the motion acceleration profile from EPICS.

File: pcdsdevices/eps.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
+"""
+Classes for EPS readbacks.
```
[the remark this answers is not in the document]
Comment (ZLLentz) [reply pcdsdevices/eps.py:2]: Good call

[the remark this answers is not in the document]
Comment (ZLLentz) [reply pcdsdevices/eps.py]: Expanded in https://github.com/pcdshub/pcdsdevices/pull/1157/commits/5ba6d1e9842d1aa7e3c3f27de10ea9d22dbe6000

File: docs/source/upcoming_release_notes/1157-api_motion_pv_cull.rst [hunk 3]
```diff
@@ -0,0 +1,43 @@
+1157 api_motion_pv_cull
+#######################
+
+API Changes
+-----------
+- Remove rarely-used TwinCAT state config PVs from `TwinCATStateConfigOne`
+  that are also being removed from the IOCs.
+  These are still available on the PLC itself.
+
+  - ``delta``
```
Comment (klauer) [inline docs/source/upcoming_release_notes/1157-api_motion_pv_cull.rst:10]: `rst` rendering is always dicey for me, but looks like you got it right here - looks good 👍

File: pcdsdevices/eps.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+    # See _update_flag docstring
+    set_metadata(flags, dict(variety="bitmask", bits=8))
+
+    @flags_raw.sub_value
+    def _update_flag(self, value, *args, **kwargs):
+        """
+        Update the flags value to a form that is usable in a pydm bitmask.
+
+        If bit 32 is TRUE, this comes as a negative number because EPICS CA has
```
Comment (klauer) [inline pcdsdevices/eps.py:51]: Not much to add on this other than - great description that'll assuredly help us once this falls out of working memory 👍

**Review by klauer — APPROVED** [review]
The Python changes look good to me.

I was hoping we'd get some more comments on the PLC library modification before I got to this PR, but it's already COB on Friday and no such luck. 

There are a few pre 1.4.0 [TwinCAT motion projects out there](https://pcdshub.github.io/plc-summary/versions.html), but live PVs - as you say - are a bit of a pain to find. I think the added override of the component in that subclass looks right.

**Pull request merged**